### PR TITLE
Feat/basic auth

### PR DESCRIPTION
**Pull request by AishwaryaKulkarni97**
# Description

Disable Basic Authentication using static passwords

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The commit message/PR follows our guidelines

# **Other information**:

### CODE REVIEW

File: jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/GKEClusterRule/DisableBasicAuthenticationRule.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            {
+                validationResult=false;
+            }
+
+        }
+
+        return validationResult;
+    }
+    // Method to compare two versions. Returns 1 if v2 is smaller, -1 if v1 is smaller, 0 if equal
+    static int versionCompare(String v1, String v2)
```
Comment (dheerajkholia-paladin) [inline jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/GKEClusterRule/DisableBasicAuthenticationRule.java:111]: To compare the version, we can parse the strings to Float and use Float.compare

File: jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/collector/GKEClusterInventoryCollector.java [hunk 2]
```diff
@@ -73,16 +72,10 @@ public List<GKEClusterVH> fetchGKEClusterInventory(ProjectVH project) throws IOE
...
+                    gkeClusterVH.setVersion(cluster.getCurrentMasterVersion());
                     gkeClusterVH.setRegion(cluster.getLocation());
                     gkeClusterVH.setRegion(cluster.getLocation());
-                    if (cluster.getMasterAuthorizedNetworksConfig() != null) {
-                        HashMap<String, Object> masterAuthorizedNetworksConfigMap = new Gson().fromJson(
-                                cluster.getMasterAuthorizedNetworksConfig().toString(),
-                                HashMap.class);
 
-                        gkeClusterVH.setMasterAuthorizedNetworksConfig(masterAuthorizedNetworksConfigMap);
-                    }
```
Comment (dheerajkholia-paladin) [inline jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/collector/GKEClusterInventoryCollector.java:85]: Please confirm once if these deletion is expected.

File: jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/GKEClusterRule/DisableBasicAuthenticationRule.java (hunk 1 above)
Comment (AishwaryaKulkarni97) [reply jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/GKEClusterRule/DisableBasicAuthenticationRule.java]: fixed logic for version comparison

File: jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/collector/GKEClusterInventoryCollector.java (hunk 2 above)
Comment (AishwaryaKulkarni97) [reply jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/collector/GKEClusterInventoryCollector.java:85]: as discussed with @AnjaliMadhavi-Nakirikanti  the code to set masterAuthorizedNetworksConfigMap was incorrect due to which GKEClusterInventoryCollector was throwing an error. Therefore, I have removed the code as of now to test this policy. However, @AnjaliMadhavi-Nakirikanti is working on rectifying code for masterAuthorizedNetworksConfigMap

**Review by dheerajkholia-paladin — APPROVED** [review]

**Review by santhosh-challa — APPROVED** [review]

**Review by santhosh-challa — APPROVED** [review]

**Pull request merged**